appcelerator/titanium_mobile
# feat(android): app-bundle support

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-26434

**Summary:**
Building an Android app for "production" will now generate a `*.aab` Android App Bundle file in addition to the `*.apk` file.

**Note:**
An app-bundle cannot be installed/ran on an Android device. It is a "publishing format" only supported by the Google Play app store. When uploaded, Google Play will generate multiple APK files split by CPU architecture and image density.

**Test:**
1. Launch Appcelerator Studio.
2. Create or open an existing Titanium app project.
3. Click on the project in the "Project Explorer" panel in studio.
4. Click on the left-most drop-down box in the toolbar and select "Package".
5. Click on the drop-down box to the right and select "Android Play Store".
6. Click on the left-most [>] run button on the toolbar.
7. A "Distribute" dialog window will appear.
8. Enter a "Distribution Location" the app's release files will be built to.
9. Enter the keystore info to digitally sign the app with.
10. Click on the "Publish" button.
11. When the build finishes, go to the "Distribution Location" directory.
12. Verify that 2 files were created. An `<AppName>.apk` and `<AppName>.aab` file. _(The AAB file is the app-bundle.)_
13. Verify that you can upload the `*.aab` and keystore to Google Play without error.
